### ISSUE TITLE
Error when adding table with time frames

### ISSUE THREAD

**Issue by constantinpape** [haesleinhuepf/napari-skimage-regionprops#41]
First of all thanks for the awesome plugin! It provides exactly the functionality I want for visualizing segmentation features in napari.
My "normal mode of operation" would be to call this directly from python using `add_table`, since this offers most flexibility to look at classification predictions from a network, clustering results etc. in addition to standard regionprops features.
This works as expected for 2d data, as explained here: https://forum.image.sc/t/exloring-image-segment-features-in-napari/75222/7?u=constantinpape

However, I run into an issue when using this for 2d data with time; here is a minimal reproducable example:
```python
import numpy as np
import napari
import pandas as pd
from skimage.data import binary_blobs
from skimage.measure import regionprops_table, label
from napari_skimage_regionprops._table import add_table

# a random image with 3 timepoints
image = np.random.rand(3, 128, 128)

# and a segmentation for the 3 timepoints
segmentation = np.stack([label(binary_blobs(128, volume_fraction=0.25)) for _ in range(3)])
print(image.shape, segmentation.shape)

# compute the features for each timepoint
features = []
for t, (seg, im) in enumerate(zip(segmentation, image)):
    feats = regionprops_table(seg, im, properties=("label", "mean_intensity"))
    # add the frame column
    feats["frame"] = np.full(len(feats["label"]), t)
    features.append(pd.DataFrame(feats))
features = pd.concat(features, axis=0)

v = napari.Viewer()
v.add_image(image)
label_layer = v.add_labels(segmentation)
label_layer.features = features
add_table(label_layer, v)
napari.run()
```

This opens napari, but when I try to select a label in the table or image nothing happens and the following error is thrown:
```
UnboundLocalError                         Traceback (most recent call last)
File ~/Work/my_projects/napari-skimage-regionprops/napari_skimage_regionprops/_table.py:120, in TableWidget._after_labels_clicked(self=<napari_skimage_regionprops._table.TableWidget object>)
    118 if frame_column is not None and self._viewer is not None:
    119     for r, (l, f) in enumerate(zip(self._table["label"], self._table[frame_column])):
--> 120         if l == self._layer.selected_label and f == frame:
        self._layer = <Labels layer 'segmentation' at 0x7f1f47ba9e80>
        self = <napari_skimage_regionprops._table.TableWidget object at 0x7f1f442def70>
        l = 12
        f = 0
    121             self._view.setCurrentCell(r, self._view.currentColumn())
    122             break

UnboundLocalError: local variable 'frame' referenced before assignment
```

**Comment by constantinpape** [comment]
Oh, and I am using the latest napari version (v0.4.17) and the current master branch of `napari-skiamge-regionprops`.

**Comment by haesleinhuepf** [comment]
Hey @constantinpape ,

I just ran your code (on nsr master) and the error doesn't happen on my side. I can fix it anyway as there is indeed a case where this variable is not set.

> 2d data with time

Be a bit careful here. Napari uses TZYX axis order. All tools we have developed so far for timelapse-analysis need to respect this. That means you should convert your dataset in a 4D dataset where size in Z = 1. Otherwise your 2D+t dataset will be interpreted as 3D dataset with no time. See also this hint in the [napari-time-slicer documentation](https://github.com/haesleinhuepf/napari-time-slicer):

> If you want to convert a 3D dataset into a 2D + time dataset, use the menu Tools > Utilities > Convert 3D stack to 2D timelapse (time-slicer). It will turn the 3D dataset to a 4D datset where the Z-dimension (index 1) has only 1 element, which will in napari be displayed with a time-slider. Note: It is recommended to remove the original 3D dataset after this conversion.

**Comment by constantinpape** [comment]
> That means you should convert your dataset in a 4D dataset where size in Z = 1.

Ok, that's good to now. I will just make sure to add the singleton z-axis in this case.

**Comment by constantinpape** [comment]
> I just ran your code (on nsr master) and the error doesn't happen on my side.

Yes, the code itself runs, but it then fails when selecting a label in napari (either in the table or the image).
But your fix in #42 looks good, and I just checked and adding the singleton z-axis also fixes the issue.

**Comment by haesleinhuepf** [comment]
Ok, cool. I'll add a comment to the documentation. Btw. you use-case also sounds like you want to give the [napari-clusters-plotter](https://github.com/BiAPoL/napari-clusters-plotter) a try ;-)

**Comment by haesleinhuepf** [comment]
I just release [version 0.6.0](https://github.com/haesleinhuepf/napari-skimage-regionprops/releases/tag/0.6.0) containing the discussed changes. Let me know if any issue persists on your side!

**Comment by constantinpape** [comment]
> I just released [version 0.6.0](https://github.com/haesleinhuepf/napari-skimage-regionprops/releases/tag/0.6.0) containing the discussed changes. Let me know if any issue persists on your side!

Will do.

> Ok, cool. I'll add a comment to the documentation. Btw. you use-case also sounds like you want to give the [napari-clusters-plotter](https://github.com/BiAPoL/napari-clusters-plotter) a try ;-)

Thanks for the pointer, that's indeed very helpful for exploring features and different clusterings!
